### PR TITLE
{Graph} Refine code and tests for Microsoft Graph User

### PR DESCRIPTION
**Pull request by calvinhzy**
**Description**

Fix tests and refactor.
Following #21525

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/role/tests/latest/test_graph.py [hunk 1]
```diff
@@ -559,25 +559,59 @@ class GraphUserScenarioTest(ScenarioTest):
...
-            checks=[self.check("displayName","{user1}")]
+            'ad user create --display-name {user1} '
+            '--mail-nickname {mail_nickname} '
+            '--password {password} '
+            '--force-change-password-next-login {force_change_password_next_login} '
+            '--user-principal-name {user1}@{domain} ',
+            checks=[
+                self.check("displayName","{user1}"),
+                self.check("userPrincipalName", "{user1}@{domain}")
+            ]
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/role/tests/latest/test_graph.py:580]: Please make sure all properties like `-mail-nickname` are also checked.

File: src/azure-cli/azure/cli/command_modules/role/tests/latest/test_graph.py [hunk 2]
```diff
@@ -559,25 +559,59 @@ class GraphUserScenarioTest(ScenarioTest):
...
+        group_result = self.cmd(
+            'ad group create --display-name {group} --mail-nickname {group} --description {group}').get_output_in_json()
+        self.kwargs['group_id'] = group_result['id']
+        # add user1 into group
+        self.cmd('ad group member add -g {group} --member-id {user1_id}',
+                 checks=self.is_empty())
+
+        # show user's group memberships
+        self.cmd('ad user get-member-groups --upn-or-object-id {user1_id}',
+                 checks=self.check('[0].displayName', self.kwargs['group']))
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/role/tests/latest/test_graph.py:614]: I think this is already covered by `group`'s test, right? @evelyn-ys

File: src/azure-cli/azure/cli/command_modules/role/tests/latest/test_graph.py [hunk 3]
```diff
@@ -559,25 +559,59 @@ class GraphUserScenarioTest(ScenarioTest):
...
-                                       checks=[self.check("displayName", '{user1}_new')]
-                                       ).get_output_in_json()
+        self.cmd('ad user show --upn-or-object-id {user1}@{domain}',
+                 checks=[
+                     self.check("displayName", '{user1_newName}')
+                 ])
+        self.cmd('ad user update --id {user1}@{domain} --password {password}')
+        self.cmd('ad user update --id {user1_id} --password {password} --force-change-password-next-login true')
+        with self.assertRaises(CLIError):
+            self.cmd('ad user update --id {user1_id} --force-change-password-next-login false')
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/role/tests/latest/test_graph.py:601]: Kindly add some comments on why it fails, or use `assertRaisesRegx` to check the error message.

File: src/azure-cli/azure/cli/command_modules/role/tests/latest/test_graph.py [hunk 4]
```diff
@@ -559,25 +559,59 @@ class GraphUserScenarioTest(ScenarioTest):
...
+        self.cmd('ad user update --id {user1_id} --password {password}')
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/role/tests/latest/test_graph.py:602]: We can do this at L587 in one shot.

**Comment by yonzhan** [comment]
Graph

File: src/azure-cli/azure/cli/command_modules/role/tests/latest/test_graph.py (hunk 2 above)
Comment (calvinhzy) [reply src/azure-cli/azure/cli/command_modules/role/tests/latest/test_graph.py:614]: I believe it has been removed from group test

File: src/azure-cli/azure/cli/command_modules/role/tests/latest/test_graph.py (hunk 1 above)
Comment (calvinhzy) [reply src/azure-cli/azure/cli/command_modules/role/tests/latest/test_graph.py:580]: --mail-nickname not returned by the result or the show command.

File: src/azure-cli/azure/cli/command_modules/role/tests/latest/test_graph.py (hunk 2 above)
Comment (evelyn-ys) [reply src/azure-cli/azure/cli/command_modules/role/tests/latest/test_graph.py:615]: This is removed from group scenario test. Group scenario only covers commands beginning with `az ad group`

**Review by jiasli — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**